pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -81,6 +81,33 @@ func writeVifFile(buf []byte, pid, name string) error {
 	return nil
 }
 
+func setPodInterfaceIPCache(iface *v1.Interface, podInterfaceName string, pid string) error {
+	link, err := Handler.LinkByName(podInterfaceName)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get a link for interface: %s", podInterfaceName)
+		return err
+	}
+	// get IP address
+	addrList, err := Handler.AddrList(link, netlink.FAMILY_V4)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get a link for interface: %s", podInterfaceName)
+		return err
+	}
+	// no ip assigned. ipam disabled
+	if len(addrList) == 0 {
+		return nil
+	}
+
+	err = writeToCachedFile(iface, interfaceCacheFile, pid, iface.Name)
```

Comment:
something worth considering. These files will need to be written to predictable locations that virt-handler's reconcile loop can discover. 

There's a directory used for this kind of ephemeral caching information that is set in the `virtPrivateDir` variable in the reconcile loop. 

I'd suggest making a network-info caching directory in there.

For example, the directory could be per vmi with UID acting as the unique id.

`/var/run/kubevirt-private/network-info-cach/<VMI UID>`

Then each interface in the pod's spec could have it's own file. For example, the default pod network that's automatically assigned would be here. 

`/var/run/kubevirt-private/network-info-cach/<VMI UID>/default`

and it would have the json contents written to the file could then look something like this. 

```
{
  name: "default",
  interface: "eth0",
  defaultIP: "10.10.0.1"
  allIPs: {
      "10.10.0.1",
       "10.10.0.2",
   }
}
```

When virt-handler's updateStatus function is setting the IP information, it can then scan for this cach info in the well known file, and look up all the ip information from disk. We migth want to make a read through cache so that files are only read once in the reconcile loop and stored in memory afterwards since we know these files will never mutate.

Virt-handler's `/processVmCleanup()` function will need to be responsible for deleting the `/var/run/kubevirt-private/network-info-cache/<VMI UID' dir during shutdown.

Reply:
> something worth considering. These files will need to be written to predictable locations that virt-handler's reconcile loop can discover.
> 
> There's a directory used for this kind of ephemeral caching information that is set in the `virtPrivateDir` variable in the reconcile loop.
> 
> I'd suggest making a network-info caching directory in there.
> 
> For example, the directory could be per vmi with UID acting as the unique id.
> 
> `/var/run/kubevirt-private/network-info-cach/<VMI UID>`
> 
> Then each interface in the pod's spec could have it's own file. For example, the default pod network that's automatically assigned would be here.
> 
> `/var/run/kubevirt-private/network-info-cach/<VMI UID>/default`
> 
> and it would have the json contents written to the file could then look something like this.
> 
> ```
> {
>   name: "default",
>   interface: "eth0",
>   defaultIP: "10.10.0.1"
>   allIPs: {
>       "10.10.0.1",
>        "10.10.0.2",
>    }
> }
> ```
The content of the file would look a little different here, because `v1.Interface` does not have IP info, had to put it into a struct with PodIP got from `Handler.AddrList`
> 
> When virt-handler's updateStatus function is setting the IP information, it can then scan for this cach info in the well known file, and look up all the ip information from disk. We migth want to make a read through cache so that files are only read once in the reconcile loop and stored in memory afterwards since we know these files will never mutate.
> 
> Virt-handler's `/processVmCleanup()` function will need to be responsible for deleting the `/var/run/kubevirt-private/network-info-cache/<VMI UID' dir during shutdown.